bokeh/models/ranges.py

```diff
@@ -90,6 +100,19 @@ class DataRange1d(DataRange):
     automatically computed end value.
     """)
 
+    bound_lower = Float(help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data. If not provided, will
+    default to the min of the data.
+    """)
```

Comment:
I disagree, I think this is one of the cases people who've been asking for this feature want to see improve.

My interpretation was that if you do the following:
```python
p = figure()
p.scatter(lots of numbers)
```
Then your plot is auto-ranged (start and end are set to min and max of your data) and if you're zooming out you can't go off into infinity.

 I could be wrong, but I think its worth at least asking around.

I think people are also looking for the converse, that you can't zoom in crazy far either and this PR doesn't handle that use case.